src/Agent.Listener/Agent.cs

```diff
@@ -151,32 +151,14 @@ public override void Initialize(IHostContext hostContext)
             finally
             {
                 _term.CancelKeyPress -= CtrlCHandler;
-                HostContext.Unloading -= Agent_Unloading;
-                _completedCommand.Set();
-            }
-        }
-
-        private void Agent_Unloading(object sender, EventArgs e)
-        {
-            if ((!_inConfigStage) && (!TokenSource.IsCancellationRequested))
-            {
-                TokenSource.Cancel();
-                _completedCommand.WaitOne(Constants.Agent.ExitOnUnloadTimeout);
             }
         }
 
         private void CtrlCHandler(object sender, EventArgs e)
         {
+            TokenSource.Cancel();
+            _term.WriteLine();
```

Comment:
think about whether this might be a thread synchronization problem (and whether warrants adding a mutex inside terminal class)